src/class-integration-composite.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace WPML\PB\Gutenberg;
+
+class Integration_Composite implements Integration {
+
+	/**
+	 * @var Integration[] $integrations
+	 */
+	private $integrations;
+
+	/**
+	 * @param Integration[] $integrations
+	 */
+	public function __construct( array $integrations ) {
+		$this->integrations = $integrations;
+	}
+
+	public function add_hooks() {
+		foreach ( $this->integrations as $integration ) {
+
+			if ( ! $integration instanceof Integration ) {
```

Comment:
@strategio I know this is strictly correct but throwing exceptions for this kind of things is redundant to me.

If you really want to make sure the class is used the correct way then I think you should not allow the class to except an array of unknown types in the first place but make the caller add the correct items.

eg.
```
class Integration_Composite implements Integration {
	/**
	 * @var Integration[] $integrations
	 */
	private $integrations = [];

	public function add( Integration $integration ) {
		$this->integrations[] = $integrations;
	}
 
...
}
```

This way the IDE will tell you're if you doing it wrong.

Reply:
done @brucepearson, I opted for the `add` method.